static/js/enketo/widgets/simprints.js

```diff
@@ -51,22 +32,28 @@ define( function( require, exports, module ) {
 
     Simprintswidget.prototype._init = function() {
         var $el = $( this.element );
-
-        $el.find( 'input' ).attr( 'disabled', true );
-
-        if ( !window.medicmobile_android ) {
-            $el.append( '<p>Simprints is only supported for android app users.</p>' );
-            return;
-        }
-        if ( !window.medicmobile_android.simprints_reg ) {
-            $el.append( '<p>Simprints is not supported on this version of the wrapper app.</p>' );
+        var $input = $el.find( 'input' );
+        $input.hide();
+        var angularServices = angular.element( document.body ).injector();
+        var $translate = angularServices.get( '$translate' );
+        var service = angularServices.get( 'Simprints' );
+
+        if ( !service.enabled() ) {
+            $translate('simprints.disabled').then(function(label) {
+                $el.append( '<p>' + label + '</p>' );
+            });
             return;
         }
 
-        /* jslint bitwise: true */
-        var simprintsInputId = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER) & SP_ID_MASK;
-        $el.find( 'input' ).attr( 'data-simprints-reg', simprintsInputId );
-        $el.append( '<button class="btn btn-primary" onclick="medicmobile_android.simprints_reg(' + simprintsInputId + ')"><i class="fa fa-bluetooth"> Get new Simprints ID</button>' );
+        $el.on('click', '.btn.simprints-register', function() {
+            service.register().then(function(response) {
+                $input.val(response.id).trigger('change');
+            });
+        });
+
+        $translate('simprints.register').then(function(label) {
+            $el.append( '<div><a class="btn btn-default simprints-register"><img src="static/img/simprints.png" width="20" height="20"/> ' + label + '</a></div>' );
```

Comment:
Sounds reasonable; I hope it works :~)  From a brief play with jQuery it looks like it would:

<img width="425" alt="screen shot 2017-08-14 at 12 51 09" src="https://user-images.githubusercontent.com/191496/29268913-494c1a38-80ef-11e7-8e95-78affd768724.png">